jmeter/plans/apply.rb

```diff
@@ -0,0 +1,92 @@
+require 'ruby-jmeter'
+
+BASEURL = ENV.fetch('JMETER_TARGET_BASEURL')
+
+def url(path)
+  BASEURL + path
+end
+
+# 975 concurrent users, ~7 minutes per session
+test do
+  cookies clear_each_iteration: true
+  thread_count = 975
+
+  threads count: thread_count, continue_forever: true, duration: 420 do
+    #-> Sign up
+    visit name: 'Account page', url: url('/candidate/account') do
+      extract name: 'authenticity_token', regex: 'name="authenticity_token" value="(.+?)"'
+    end
+    think_time 2000
+    submit(
+      name: 'Account page - create an account', url: url('/candidate/account'),
+      fill_in: {
+        'candidate_interface_create_account_or_sign_in_form[existing_account]' => 'false',
+        'authenticity_token' => '${authenticity_token}',
+      }
+    )
+    think_time 1000
+    visit name: 'Sign up page', url: url('/candidate/sign-up') do
+      extract name: 'authenticity_token', regex: 'name="authenticity_token" value="(.+?)"'
+    end
+    think_time 2000
+    submit(
+      name: 'Sign up page - submit email', url: url('/candidate/account'),
+      fill_in: {
+        'authenticity_token': '${authenticity_token}',
+        'candidate_interface_sign_up_form[email_address]': '${__threadNum}' + '@loadtest.example.com',
+        'candidate_interface_sign_up_form[accept_ts_and_cs][]': '',
+        'candidate_interface_sign_up_form[accept_ts_and_cs]': 'true',
+        'commit': 'Continue',
+      }
+    )
+    think_time 1000
+
+    #-> Sign in
+    visit name: 'Account page', url: url('/candidate/account') do
+      extract name: 'authenticity_token', regex: 'name="authenticity_token" value="(.+?)"'
+    end
+    think_time 2000
+    submit(
+      name: 'Account page - sign in', url: url('/candidate/account'),
+      fill_in: {
+        'candidate-interface-create-account-or-sign-in-form-email-field' => '${__threadNum}' + '@loadtest.example.com',
+        'authenticity_token' => '${authenticity_token}',
+      }
+    )
+    think_time 1000
+    # TODO: add a bypass to our auth logic that immediately signs in @loadtest.example.com email addresses
```

Comment:
So would this basically just call all the sign up actions then redirect to the auth path with the token?

Reply:
Yes, I think so. Would need to another step in that case which confirms the sign in.